OneCommunityGlobal/HighestGoodNetworkApp
# Jeffrey>Tzu Ning correct fix letter code dropdown

# Description
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/112568110/0de3ab8f-4b74-4d1b-9c30-5d8fe18957d2)

## Related PRS (if any):
none
…

## Main changes explained:
- Add a handler function that fires when the new team code has been accepted and sent to database
- Said handler function updates state (teamCodes, selectedCodes, and summaries) within WeeklySummariesReport.jsx to accurately reflect the changes within that would occur if updated data were pulled from the database, but without refreshing

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. log in as Admin/Owner user
4. go to Other Links → Reports → Weekly Summaries Reports
5. select a team code from the dropdown that has a label of (1)
6. change the team's code (input element next to Media URL) to a code that does not exist yet
7. ensure that the initial team code from step 5 is not present within the dropdown menu
8. click on the dropdown again and see if that new team code is reflected in the dropdown menu
9. select that team code to ensure the correct number of members are populated on the screen
10. select another team code
11. change the team's code to a code that DOES exist already
12. click on the dropdown and check that the team code from step 11 is updated correctly
13. select that team code to ensure the correct number of members are populated on the screen
14. refresh page to confirm that the changes you made previously (and saw accurately reflected on the front-end before the refresh) are the same after you refresh

**Note** - this is where you can find the team code (example below is C-SOC):
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/112568110/5986bb94-02c2-45c8-8ba0-8a7e5f66ebed)

## Screenshots or videos of changes:

Before change: 
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/112568110/5617d750-d275-40d3-8a95-f787b5a30c61

After change:
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/112568110/3b30d14c-2207-455a-819d-2fc9be638131
